closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -416,16 +469,22 @@ int decryptFinal(byte[] in, int inOfs, int len,
                      byte[] out, int outOfs)
         throws IllegalBlockSizeException, AEADBadTagException,
         ShortBufferException {
-        if (len < tagLenBytes) {
+        if (len < 0) {
+            throw new ProviderException("Input length is negative");
+        }
+        if (len < tagLenBytes - ibuffer.size()) {
             throw new AEADBadTagException("Input too short - need tag");
         }
+        if (len > MAX_BUF_SIZE - ibuffer.size()) {
+            throw new ProviderException("SunJCE provider only supports "
+                + "a positive input size up to " + MAX_BUF_SIZE + " bytes");
+        }
 
-        // do this check here can also catch the potential integer overflow
-        // scenario for the subsequent output buffer capacity check.
-        checkDataLength(ibuffer.size(), (len - tagLenBytes));
-
+        if (ibuffer.size() < tagLenBytes - len) {
+            throw new AEADBadTagException("Input too short - need tag");
+        }
```

Comment:
This is redundant (a repeat of the test on line 475).

Reply:
Removed.